src/connector/builder.rs

```diff
@@ -237,6 +240,22 @@ impl ConnectorBuilder<WantsProtocols1> {
         })
     }
 
+    /// Override server name for the TLS stack
+    ///
+    /// By default, for each connection hyper-rustls will extract host portion
+    /// of the destination URL and verify that server certificate contains
+    /// this value.
+    ///
+    /// If this method is called, hyper-rustls will instead use this resolver
+    /// to compute the value used to verify the server certificate.
+    pub fn with_server_name_resolver<T>(mut self, resolver: T) -> Self
```

Comment:
Nit: let's spell this as `impl ResolveServerName + Send + Sync + 'static`, removing the intermediate type argument name (and I think the order of traits is more conventional this way).